crates/client/src/h2_client_connection.rs

```diff
@@ -105,18 +113,21 @@ impl<T> HttpsClientConnection<T> {
     /// * `name_server` - IP and Port for the remote DNS resolver
     /// * `bind_addr` - IP and port to connect from
     /// * `dns_name` - The DNS name, Subject Public Key Info (SPKI) name, as associated to a certificate
+    /// * `http_endpoint` - The HTTP endpoint where the resolver provides service, typically `/dns-query`
     /// * `client_config` - The TLS config
     #[allow(clippy::new_ret_no_self)]
     pub fn new_with_bind_addr(
         name_server: SocketAddr,
         bind_addr: Option<SocketAddr>,
         dns_name: String,
+        http_endpoint: String,
```

Comment:
Suggest we also make this `Arc<str>`, maybe it should even become `Option<Arc<str>>` so we can default to `/dns-query`?

Reply:
I think `http_endpoint` goes hand in hand with `dns_name`, so I made it a `String`. Should I change both to `Arc<str>`?

As for Option... I am having a hard time determining the exact level of abstraction where `Option<String>` should become `String` (ie where None is substituted by `/dns-query`). `hickory_proto` feels way too low level for this and I think we should try to avoid any magicalness here.

Reply:
> I think `http_endpoint` goes hand in hand with `dns_name`, so I made it a `String`. Should I change both to `Arc<str>`?

I think we should make that change, but it's sort of off-topic for this PR. Maybe make a separate PR for it, or a separate commit in this PR?

> As for Option... I am having a hard time determining the exact level of abstraction where `Option<String>` should become `String` (ie where None is substituted by `/dns-query`). `hickory_proto` feels way too low level for this and I think we should try to avoid any magicalness here.

That's fair.